src/main/java/com/school/project/service/impl/LessonServiceimpl.java

```diff
@@ -0,0 +1,68 @@
+package com.school.project.service.impl;
+
+import com.school.project.exception.LessonNotFoundException;
+import com.school.project.model.entities.Lesson;
+import com.school.project.repository.LessonRepository;
+import com.school.project.service.LessonService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class LessonServiceimpl implements LessonService {
+
+    @Autowired
+    LessonRepository lessonRepository;
+
+    @Override
+    public Lesson createLesson(Lesson lesson) {
+        return lessonRepository.save(lesson);
+    }
+
+    @Override
+    public Lesson getLesson(Long id) {
+        Optional<Lesson> byId = lessonRepository.findById(id);
+        if (!byId.isPresent()) throw new LessonNotFoundException("Lesson with id : " + id + " is not found");
+        return byId.get();
+    }
+
+    @Override
+    public List<Lesson> getAllLessonsByGroup(Long id) {
+        List<Lesson> lessonByGroupId = lessonRepository.getAllByGroupId(id);
+        validationResponseIsNotEmpty(lessonByGroupId,"Group",id);
```

Comment:
pls use: 
For windows Ctrl+Alt+L.
For ubuntu Ctrl+Alt+windows+L.
For Mac Command+Option+L.